src/components/UniversalNavbar/UniversalNavbar.js

```diff
@@ -73,218 +73,209 @@ const LINKS = {
   ],
 }
 
-class UniversalNavbar extends React.Component {
-  state = {
-    showMobileMenu: false,
-  }
+const UniversalNavbar = ({
+  LinkComponent,
+  hideMobileCta,
+  hideDesktopCta,
+  logoHref,
+  trackCtaClick,
+}) => {
+  const [showMobileMenu, setShowMobileMenu] = useState(false)
 
-  toggleHamburger = () => {
-    // TODO: use functional setState
-    this.setState({ showMobileMenu: !this.state.showMobileMenu })
+  const toggleHamburger = () => {
+    setShowMobileMenu(!showMobileMenu)
   }
 
-  handleHamburgerKeyPress = (event) => {
+  const handleHamburgerKeyPress = (event) => {
     if (event.key === 'Enter') {
-      this.toggleHamburger()
+      toggleHamburger()
     }
   }
 
-  render() {
-    const {
-      LinkComponent,
-      hideMobileCta,
-      hideDesktopCta,
-      logoHref,
-    } = this.props
+  const renderCtaButton = (showWhenScrolled) => (
+    <a
+      className={'cta-button ' + (showWhenScrolled ? 'show-when-scrolled' : '')}
+      onClick={trackCtaClick}
+      href={LINKS.TERM.href}
+    >
+      <Button.Small.BlackOutline>Check my price</Button.Small.BlackOutline>
+    </a>
+  )
 
-    const renderCtaButton = (showWhenScrolled) => (
-      <a
-        className={
-          'cta-button ' + (showWhenScrolled ? 'show-when-scrolled' : '')
-        }
-        onClick={this.props.trackCtaClick}
-        href={LINKS.TERM.href}
+  const renderTextLink = (link) => (
+    <div key={link.id} className={styles.textLink}>
+      <NavLink
+        href={link.href}
+        LinkComponent={link.href !== '/login/' ? LinkComponent : null}
```

Comment:
This seems a little convoluted to me. Is there a cleaner way we can do this?

Reply:
Should be fixed in this: https://github.com/getethos/ethos-design-system/pull/275/commits/bfbb5a3f0520f2f157c895f155212546bf09b3a7